### PR TITLE
Provides backend for unique referral code and shareable links

### PR DESCRIPTION
**Pull request by SaptakS**
##### Description
- Backend to create a referral code
- Backend to get profile from referral code
- Backend to create uniquely shareable bounty link
- Backend to get bounty and inviter from the unique link

refs: #3950

##### Checklist

- [ ] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [ ] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

###### Contributor

- [ ] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Tested all changes **locally**
- [ ] Verified existing functionality
- [ ] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@SaptakS would it make sense to add tests for these ?

File: app/dashboard/utils.py [hunk 1]
```diff
@@ -273,6 +274,40 @@ def get_web3(network, sockets=False):
...
+
+
+def get_bounty_from_invite_url(invite_url):
+    """Returns a unique url for each bounty and one who is inviting
+
+    Returns:
+        A unique string for each bounty
+    """
+    salt = "X96gRAVvwx52uS6w4QYCUHRfR3OaoB"
+    decoded_string = base64.urlsafe_b64decode(string.encode()).decode()
```
Comment (thelostone-mc) [inline app/dashboard/utils.py]: i like

**Comment by SaptakS** [comment]
@thelostone-mc I am not sure since this isn't integrated with a frontend as to what exactly I should write in tests. Just the string output? That would almost be like testing base64 package than what we are doing? Do you have anything in mind?

**Comment by thelostone-mc** [comment]
Ah cool I guess 🤷‍♂️

**Pull request merged**